src/controllers/highlights/HighlightController.cpp

```diff
@@ -167,7 +167,16 @@ void rebuildUserHighlights(Settings &settings,
                 (void)args;             // unused
                 (void)badges;           // unused
                 (void)originalMessage;  // unused
-                (void)self;             // unused
+
+                if (self && getSettings()->enableSelfMessagesHighlight)
+                {
+                    auto customColor =
+                        getSettings()->selfMessagesHighlightColor;
+                    return HighlightResult{
+                        false, false, (QUrl) nullptr,
+                        std::make_shared<QColor>(customColor),
+                        getSettings()->showSelfMessagesHighlightInMentions};
+                }
```

Comment:
This check should be done outside of the for loop and, if it's enabled, add itself as a user highlight check. See how it's done above in the `rebuildMessageHighlights` function for `settings.enableSelfHighlight`

Reply:
done in [a38de16](https://github.com/Chatterino/chatterino2/pull/3833/commits/a38de16a235b86b185d5da37799b2017b8425707)